Automattic/sensei
# Build JavaScript using Webpack

Fixes #2194.

This PR uses Webpack instead of Gulp to build the JS files. This has a couple of implications:
- We need to create the build using npm instead of Gulp commands. `npm run build` will run the tests and create the .zip file. `npm run build-dev` will just create the .zip file. (Note: I'll update the relevant Field Guide page once this PR is merged.)
- When developing locally, we won't be able to copy the entire Sensei folder to our local WordPress install (for those of us who do that). Instead, we'll either need to copy the entire `build/woothemes-sensei` folder, or add `define( 'SCRIPT_DEBUG', true );` to the `wp-config.php` file, so that non-minified JS files are used.

Note that Webpack was previously set up for blocks. I've commented that code out for now so that we will have a reference point for including blocks in the build process once we actually have some.

## Testing
1. Execute `npm run build-dev`.
2. Copy the `build/woothemes-sensei` folder to your WordPress install.
3. Navigate to the _Lessons_ page.
4. Ensure there is no error in the Console as described in #2194.
5. Quick edit a lesson and ensure that it works.
6. Click around the back-end and front-end with the Console open and ensure that there aren't any errors about missing JS files.
